zerver/data_import/mattermost.py

```diff
@@ -0,0 +1,540 @@
+
+        stream = build_stream(
+            date_created=now,
+            realm_id=realm_id,
+            name=channel_dict['display_name'] or channel_dict['name'],
+            description=channel_dict["purpose"] or channel_dict['header'],
+            stream_id=stream_id,
+            deactivated=False,
+            invite_only=invite_only,
+        )
+
+        channel_users = set()
+        for username in channel_admins_map[stream_name]:
+            channel_users.add(user_id_mapper.get(username))
+
+        for username in channel_members_map[stream_name]:
+            channel_users.add(user_id_mapper.get(username))
+
+        if channel_users:
+            subscriber_handler.set_info(
+                stream_id=stream_id,
+                users=channel_users,
+            )
+        streams.append(stream)
+
+    return streams
+
+def make_user_messages(zerver_message: List[ZerverFieldsT],
+                       subscriber_map: Dict[int, Set[int]],
+                       is_pm_data: bool,
+                       mention_map: Dict[int, Set[int]]) -> List[ZerverFieldsT]:
+
+    zerver_usermessage = []
+
+    for message in zerver_message:
+        message_id = message['id']
+        recipient_id = message['recipient']
+        sender_id = message['sender']
+        mention_user_ids = mention_map[message_id]
+        subscriber_ids = subscriber_map.get(recipient_id, set())
+        user_ids = subscriber_ids | {sender_id}
+
+        for user_id in user_ids:
+            is_mentioned = user_id in mention_user_ids
+            user_message = build_user_message(
+                user_id=user_id,
+                message_id=message_id,
+                is_private=is_pm_data,
+                is_mentioned=is_mentioned,
+            )
+            zerver_usermessage.append(user_message)
+
+    return zerver_usermessage
+
+def process_raw_message_batch(realm_id: int,
+                              raw_messages: List[Dict[str, Any]],
+                              subscriber_map: Dict[int, Set[int]],
+                              user_id_mapper: IdMapper,
+                              user_handler: UserHandler,
+                              get_recipient_id: Callable[[ZerverFieldsT], int],
+                              is_pm_data: bool,
+                              output_dir: str) -> None:
+
+    def fix_mentions(content: str, mention_user_ids: Set[int]) -> str:
+        for user_id in mention_user_ids:
+            user = user_handler.get_user(user_id=user_id)
+            mattermost_mention = '@{short_name}'.format(**user)
+            zulip_mention = '@**{full_name}**'.format(**user)
+            content = content.replace(mattermost_mention, zulip_mention)
+
+        content = content.replace('@channel', '@**all**')
+        content = content.replace('@all', '@**all**')
+        # We don't have an equivalent for Mattermost's @here mention which mentions all users
+        # online in the stream.
```

Comment:
Probably worth doing a `content = content.replace('@here', '@**all**')` like we have in hipchat for this case; it'll be more discoverable if/when we do add that.